src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -230,14 +230,17 @@ private[internal] trait GlbLubs {
   def weakLub(tps: List[Type]): Type = (
     if (tps.isEmpty)
       NothingTpe
-    else if (tps forall isNumericValueType)
+    else if (shouldUseNumericLub(tps))
       numericLub(tps)
     else if (tps exists typeHasAnnotations)
       annotationsLub(lub(tps map (_.withoutAnnotations)), tps)
     else
       lub(tps)
   )
 
+  private def shouldUseNumericLub(tps: List[Type]) =
+    tps forall (tp => isNumericValueType(tp.widen))
```

Comment:
Would `deconst` be sufficient? For constant types, `widen` does the same thing. Not sure of any case where where `isNumericValueType(tp.widen)` and `isNumericValueType(tp.deconst)` would differ, but the latter is both more intention revealing and cautious.

Both questions (widen vs deconst, internally or externally) are worth thinking about this in the context of the [proposed support](https://github.com/scala/scala/pull/5310) for allowing constant types to be directly expressed in the language and for them to survive through type inference where appropriate. In particular, that PR changes `isNumericValueType` to [internally widen](https://github.com/scala/scala/pull/5310/files#diff-2b1bf15b53df94495f7155a69f822adeL1417).

/cc @milessabin